### PR TITLE
EG-832-1, backlink

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/repositories/SessionRepository.scala [hunk 1]
```diff
@@ -142,4 +172,24 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
...
+  private def savePageHistory(key: String, pageHistory: List[String]): Future[RequestOutcome[Unit]] =
+    findAndUpdate(
+      Json.obj("_id" -> key),
+      Json.obj("$set" -> Json.obj(ttlExpiryFieldName -> Json.obj("$date" -> Instant.now().toEpochMilli), s"pageHistory" -> pageHistory))
+    ).map { result =>
+        result
+          .result[DefaultSessionRepository.SessionProcess]
+          .fold {
+            logger.warn(
+              s"Attempt to savePageHistory using _id=$key eturned no result, lastError ${result.lastError}"
```
Comment (woollett-r) [inline app/repositories/SessionRepository.scala:184]: Spelling eturned

File: app/repositories/SessionRepository.scala [hunk 2]
```diff
@@ -93,15 +93,45 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
...
+                  fetchNewObject = false
+    ).flatMap { r =>
+          r.result[DefaultSessionRepository.SessionProcess]
           .fold {
-            logger.warn(s"Attempt to retrieve cached process from session repo with _id=$key returned no result, lastError ${result.lastError}")
-            Left(NotFoundError): RequestOutcome[ProcessContext]
-          }(r => Right(ProcessContext(r.process, r.answers)))
+            logger.warn(s"Attempt to retrieve cached process from session repo with _id=$key returned no result, lastError ${r.lastError}")
+            Future.successful(Left(NotFoundError): RequestOutcome[ProcessContext])
+          }{ sp =>
```
Comment (woollett-r) [inline app/repositories/SessionRepository.scala:108]: IntelliJ is reporting cyclomatic complexity is too high for the method get. Perhaps this section could be moved to a separate method.

File: app/views/standard_page.scala.html [hunk 3]
```diff
@@ -18,9 +18,9 @@
 
 @this(layout: main_layout)
 
-@(page: StandardPage, ctx: PageContext)(implicit request: Request[_], messages: Messages)
+@(page: StandardPage, ctx: PageContext, backUrl: Option[String] = None)(implicit request: Request[_], messages: Messages)
```
Comment (woollett-r) [inline app/views/standard_page.scala.html:21]: Do we need the backUrl here?

Comment (cmrhmrc) [reply app/views/standard_page.scala.html:21]: no removed

File: app/repositories/SessionRepository.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/repositories/SessionRepository.scala:184]: fixed

File: app/repositories/SessionRepository.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/repositories/SessionRepository.scala:113]: Fixed

**Pull request merged**